recipes/newmat/all/conanfile.py

```diff
@@ -0,0 +1,87 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, copy, get, rmdir, save, load
+from conan.tools.microsoft import is_msvc
+import os
+
+required_conan_version = ">=1.53.0"
+
+
+class NewmatConan(ConanFile):
+    name = "newmat"
+    package_type = "library"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://newmat.net"
+    description = "Manipulate a variety of types of matrices using standard matrix operations."
+    settings = "os", "compiler", "build_type", "arch"
+    topics = ("newmat", "matrix")
+    license = "LicenseRef-newmat"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_c_subscripts": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_c_subscripts": False,
+    }
+
+    def export_sources(self):
+        copy(self, "CMakeLists.txt", self.recipe_folder, self.export_sources_folder)
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate_build(self):
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} does not support dynamic library with msvc.")
```

Comment:
Weird that this has a `shared` check. Does this maybe belong to `validate()` instead of `validate_build()`?

Reply:
corrected that now